app/controllers/sp_components_controller.rb

```diff
@@ -57,6 +58,27 @@ def destroy
     redirect_to admin_path(anchor: component&.component_type)
   end
 
+  def associate_service
+    is_component_present = SpComponent.exists?(params[:sp_component_id])
+    service = Service.find_by_id(params[:service_id])
+
+    if is_component_present && service.present?
+      @event = AssignSpComponentToServiceEvent.create(service: service, sp_component_id: params[:sp_component_id])
+
+      if @event.valid?
+        redirect_to sp_component_path(params[:sp_component_id])
+      else
+        Rails.logger.info(@event.errors.full_messages)
+
+        render :show
+      end
+    else
+      flash[:error] = I18n.t('service.errors.unknown_component') unless is_component_present
+      flash[:error] = I18n.t('services.errors.unknown_service') unless service.present?
```

Comment:
this might be a bit misleading, if both are missing only the last error will show, since `flash[:error]` gets overwritten. simplifying to a single error should be fine... again, this is our admin stuff, so we don't need to be super-granular about errors